src/main/java/duke/DoneCommand.java

```diff
@@ -0,0 +1,43 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Controls logic of completing tasks.
+ */
+class DoneCommand extends Command {
+
+    /** Index of task to be done. */
+    private int index;
+
+    DoneCommand(int index) {
+        this.index = index;
+    }
+
+
+    /**
+     * Executes completing tasks.
+     *
+     * @param tasks Stores task list.
+     * @param ui Handles user interaction.
+     * @param storage Handles input output to hard disk.
+     * @throws DukeException When I/O error occurs.
+     */
+    @Override
+    void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```

Comment:
I noticed a lot of your methods are marked as package private, why don't mark them as protected instead?